gordo_components/server/server.py

```diff
@@ -117,6 +120,36 @@ def build_app():
     def _start_timer():
         g.start_time = timeit.default_timer()
 
+    @app.before_request
+    def _set_revision_and_collection_dir():
+        g.collection_dir = os.environ[
+            current_app.config["MODEL_COLLECTION_DIR_ENV_VAR"]
+        ]
+        g.current_revision = os.path.basename(g.collection_dir)
+
+        # If a specific revision was requested, update collection_dir
+        g.revision = request.args.get("revision") or request.headers.get("revision")
```

Comment:
Good point. Always walking that line of being too strict.. I suppose since we return the revision which was used, it's _maybe_ the user's fault if two were supplied? :man_shrugging: